### ISSUE TITLE
Spadina beacon will not initialize ENR unless it's been run with medalla before.

### ISSUE THREAD

**Issue by yorickdowne** [sigp/lighthouse#1673]
## Description

A beacon set to listen to spadina will not bring its ports up, no ENR entries, if it is run "from scratch".

This can be resolved by running it with medalla first, then switching to spadina. Output below.

## Version

- Lighthouse/v0.2.12-f1180a89
- Rust 1.46

## Output

```
 lighthouse@c60e3f7d1f01:/$ lighthouse bn --http --testnet spadina --port 30323 --eth1-endpoint https://goerli.infura.io/v3/USERID --http-address 0.0.0.0 --datadir /var/lib/lighthouse
Sep 26 14:57:33.034 WARN Ethereum 2.0 is pre-release. This software is experimental.
Sep 26 14:57:33.034 INFO Lighthouse started                      version: Lighthouse/v0.2.12-f1180a89
Sep 26 14:57:33.034 INFO Configured for testnet                  name: spadina
Sep 26 14:57:33.034 INFO Data directory initialised              datadir: /var/lib/lighthouse
Sep 26 14:57:33.048 INFO Waiting for eth2 genesis from eth1      deposit_contract: 0x48b597f4b53c21b48ad95c7256b49d1779bd5890, contract_deploy_block: 3384340, eth1_endpoint: https://goerli.infura.io/v3/USERID, service: beacon
Sep 26 14:57:33.048 INFO Importing eth1 deposit logs             service: beacon
Sep 26 14:57:34.689 INFO Waiting for more deposits               valid_deposits: None, total_deposits: 0, min_genesis_active_validators: 1024, service: beacon
^C
lighthouse@c60e3f7d1f01:/$ lighthouse bn --http --testnet medalla --port 30323 --eth1-endpoint https://goerli.infura.io/v3/USERID --http-address 0.0.0.0 --datadir /var/lib/lighthouse
Sep 26 14:57:48.629 WARN Ethereum 2.0 is pre-release. This software is experimental.
Sep 26 14:57:48.632 INFO Lighthouse started                      version: Lighthouse/v0.2.12-f1180a89
Sep 26 14:57:48.632 INFO Configured for testnet                  name: medalla
Sep 26 14:57:48.633 INFO Data directory initialised              datadir: /var/lib/lighthouse
Sep 26 14:57:48.665 INFO Starting from known genesis state       service: beacon
Sep 26 14:57:48.827 INFO Block production enabled                method: json rpc via http, endpoint: https://goerli.infura.io/v3/USERID
Sep 26 14:57:49.299 INFO Beacon chain initialized                head_slot: 0, head_block: 0xebe4…660c, head_state: 0xaf6a…f64e, service: beacon
Sep 26 14:57:49.299 INFO Timer service started                   service: node_timer
Sep 26 14:57:49.300 INFO Libp2p Service                          peer_id: 16Uiu2HAmAFo7ZX8cKJpxusfEWnHEfMa2bR7jtbtPF5RupafDGwP2, service: libp2p
Sep 26 14:57:49.300 INFO ENR Initialised                         tcp: Some(30323), udp: None, ip: None, id: 0xe5dd..5652, seq: 1, enr: enr:-KO4QE4bAmLwhcuKxRM5pxMh9K9znth8Xm_-NN6xb43HnfpGNluWyKVePpLx9mYbhqxErInzYQ-AQAH-VX4mp6lQSloBh2F0dG5ldHOIAAAAAAAAAACEZXRoMpDnp11aAAAAAf__________gmlkgnY0iXNlY3AyNTZrMaEC3FeP6hc53sTX9lz4RFxiv21TbxdAqkOVPIuoODTK5S2DdGNwgnZz, service: libp2p
Sep 26 14:57:49.301 INFO Listening established                   address: /ip4/0.0.0.0/tcp/30323/p2p/16Uiu2HAmAFo7ZX8cKJpxusfEWnHEfMa2bR7jtbtPF5RupafDGwP2, service: libp2p
Sep 26 14:57:49.302 INFO HTTP API started                        port: 5052, address: 0.0.0.0, service: http
Sep 26 14:57:51.102 INFO Sync state updated                      new_state: Syncing Finalized Chain, old_state: Stalled, service: sync
^CSep 26 14:57:55.174 INFO Shutting down..
Sep 26 14:57:55.174 WARN Discovery query failed                  error: ChannelFailed("channel closed"), service: libp2p
Sep 26 14:57:55.174 INFO HTTP service shutdown                   service: http
Sep 26 14:57:55.174 INFO Saved DHT state                         service: network
Sep 26 14:57:55.174 INFO Network service shutdown                service: network
Sep 26 14:57:55.177 INFO Saved beacon chain to disk              service: beacon
^[[Alighthouse@c60e3f7d1f01:/$ lighthouse bn --http --testnet spadina --port 30323 --eth1-endpoint https://goerli.infura.io/v3/USERID --http-address 0.0.0.0 --datadir /var/lib/lighthouse
Sep 26 14:58:05.813 WARN Ethereum 2.0 is pre-release. This software is experimental.
Sep 26 14:58:05.813 INFO Lighthouse started                      version: Lighthouse/v0.2.12-f1180a89
Sep 26 14:58:05.813 INFO Configured for testnet                  name: spadina
Sep 26 14:58:05.813 INFO Data directory initialised              datadir: /var/lib/lighthouse
Sep 26 14:58:05.829 INFO Starting beacon chain                   method: resume, service: beacon
Sep 26 14:58:06.264 INFO Block production enabled                method: json rpc via http, endpoint: https://goerli.infura.io/v3/USERID
Sep 26 14:58:06.280 INFO Beacon chain initialized                head_slot: 0, head_block: 0xebe4…660c, head_state: 0xaf6a…f64e, service: beacon
Sep 26 14:58:06.280 INFO Timer service started                   service: node_timer
Sep 26 14:58:06.282 INFO Libp2p Service                          peer_id: 16Uiu2HAmAFo7ZX8cKJpxusfEWnHEfMa2bR7jtbtPF5RupafDGwP2, service: libp2p
Sep 26 14:58:06.282 INFO ENR Initialised                         tcp: Some(30323), udp: None, ip: None, id: 0xe5dd..5652, seq: 2, enr: enr:-KO4QIQ4GJiKWUos5Bf9oJeNb91yml3GFObC9LtmjQGzPkF9QgBy2NCgIIy9J2U6CW89ZREG1s-W7y3oWhd9ejGZvl8Ch2F0dG5ldHOIAAAAAAAAAACEZXRoMpDTFw37AAAAAv__________gmlkgnY0iXNlY3AyNTZrMaEC3FeP6hc53sTX9lz4RFxiv21TbxdAqkOVPIuoODTK5S2DdGNwgnZz, service: libp2p
Sep 26 14:58:06.284 INFO Listening established                   address: /ip4/0.0.0.0/tcp/30323/p2p/16Uiu2HAmAFo7ZX8cKJpxusfEWnHEfMa2bR7jtbtPF5RupafDGwP2, service: libp2p
Sep 26 14:58:06.286 INFO HTTP API started                        port: 5052, address: 0.0.0.0, service: http
^CSep 26 14:58:07.383 INFO Shutting down..
Sep 26 14:58:07.384 WARN Discovery query failed                  error: ChannelFailed("channel closed"), service: libp2p
Sep 26 14:58:07.384 INFO Saved DHT state                         service: network
Sep 26 14:58:07.384 INFO Network service shutdown                service: network
Sep 26 14:58:07.384 INFO HTTP service shutdown                   service: http
Sep 26 14:58:07.387 INFO Saved beacon chain to disk              service: beacon
```

**Comment by yorickdowne** [comment]
Update: I've been told the first behavior, won't start ENR, is expected. Second behavior, "will start ENR if in datadir that was previously another chain", might be a corner case bug still.

It looks like the beacon, if started with spadina, waits for genesis, but if started in a directory that previously had medalla, it'll throw

```
beacon_1            | Sep 26 20:37:58.582 WARN Skipping more than an epoch             request_slot: 383872, head_slot: 0, service: beacon
beacon_1            | Sep 26 20:38:02.838 WARN Low peer count                          peer_count: 0, service: slot_notifier
beacon_1            | Sep 26 20:38:02.838 INFO Searching for peers                     current_slot: 383889, head_slot: 0, finalized_epoch: 0, finalized_root: 0x0000…0000, peers: 0, service: slot_notifier
beacon_1            | Sep 26 20:38:08.123 WARN Skipping more than an epoch             request_slot: 383872, head_slot: 0, service: beacon
beacon_1            | Sep 26 20:38:10.594 WARN Skipping more than an epoch             request_slot: 383872, head_slot: 0, service: beacon
```

**Comment by yorickdowne** [comment]
After further discussion, this will be resolved in 0.3 with the directory restructure. Closing.
